packages/next/pages/_app.tsx

```diff
@@ -69,47 +75,47 @@ const warnUrl = execOnce(() => {
   }
 })
 
-export function createUrl (router) {
+export function createUrl(router: Router) {
   // This is to make sure we don't references the router object at call time
   const { pathname, asPath, query } = router
   return {
-    get query () {
+    get query() {
       warnUrl()
       return query
     },
-    get pathname () {
+    get pathname() {
       warnUrl()
       return pathname
     },
-    get asPath () {
+    get asPath() {
       warnUrl()
       return asPath
     },
     back: () => {
       warnUrl()
       router.back()
     },
-    push: (url, as) => {
+    push: (url: string, as?: string) => {
       warnUrl()
       return router.push(url, as)
     },
-    pushTo: (href, as) => {
+    pushTo: (href: string, as?: string) => {
       warnUrl()
-      const pushRoute = as ? href : null
+      const pushRoute = as ? href : ''
```

Comment:
Why was this switched from `null`?

Reply:
@Timer To keep it with the same type used by `href`, which is always a `string` or `undefined`